File: examples/python/apps.py
import requests
import os

API_KEY = os.environ.get('COGNIMA_API_KEY', 'ck_your_api_key')
BASE_URL = 'https://cog2.cognima.com.br/api/v1'

def search_all_stores(query: str, num: int = 10, country: str = 'br'):
    """Pesquisar apps em ambas as lojas (Google Play + App Store)"""
    try:
        response = requests.get(
            f'{BASE_URL}/apps/search',
            params={'q': query, 'num': num, 'country': country},
            headers={'Authorization': f'Bearer {API_KEY}'}
        )
        response.raise_for_status()
        
        data = response.json()['data']
        print(f'📱 Pesquisa de Apps: "{data["query"]}"\n')
        
        print('=== Google Play Store ===')
        for i, app in enumerate(data['playStore'], 1):
            print(f'{i}. {app["title"]}')
            print(f'   Developer: {app["developer"]}')
            print(f'   Score: {app["score"]} | {app["price"]}')
            print(f'   ID: {app["appId"]}')
            print('')
        
        print('=== Apple App Store ===')
        for i, app in enumerate(data['appStore'], 1):
            print(f'{i}. {app["title"]}')
            print(f'   Developer: {app["developer"]}')
            price = 'Free' if app.get('free') else app.get('price', 'N/A')
            print(f'   Score: {app["score"]} | {price}')
            print(f'   ID: {app["appId"]}')
            print('')
        
        return data
    except requests.exceptions.RequestException as e:
        print(f'❌ Erro: {e}')

def search_playstore(query: str, num: int = 10, country: str = 'br', lang: str = 'pt'):
    """Pesquisar apenas na Google Play Store"""
    try:
        response = requests.get(
            f'{BASE_URL}/apps/playstore',
            params={'q': query, 'num': num, 'country': country, 'lang': lang},
            headers={'Authorization': f'Bearer {API_KEY}'}
        )
        response.raise_for_status()
        
        data = response.json()['data']
        print(f'🤖 Google Play Store: "{data["query"]}"\n')
        
        for i, app in enumerate(data['results'], 1):
            print(f'{i}. {app["title"]}')
            print(f'   {app["developer"]} | ⭐ {app["score"]} | {app["price"]}')
            summary = app.get('summary', '')[:60] if app.get('summary') else ''
            print(f'   {summary}...')
            print('')
        
        return data
    except requests.exceptions.RequestException as e:
        print(f'❌ Erro: {e}')

def search_appstore(query: str, num: int = 10, country: str = 'br'):
    """Pesquisar apenas na Apple App Store"""
    try:
        response = requests.get(
            f'{BASE_URL}/apps/appstore',
            params={'q': query, 'num': num, 'country': country},
            headers={'Authorization': f'Bearer {API_KEY}'}
        )
        response.raise_for_status()
        
        data = response.json()['data']
        print(f'🍎 Apple App Store: "{data["query"]}"\n')
        
        for i, app in enumerate(data['results'], 1):
            print(f'{i}. {app["title"]}')
            price = 'Free' if app.get('free') else app.get('price', 'N/A')
            print(f'   {app["developer"]} | ⭐ {app["score"]} | {price}')
            genres = ', '.join(app.get('genres', [])) if app.get('genres') else 'N/A'
            print(f'   Genres: {genres}')
            print('')
        
        return data
    except requests.exceptions.RequestException as e:
        print(f'❌ Erro: {e}')

def get_app_details(app_id: str, store: str = 'playStore', country: str = 'br'):
    """Obter detalhes de um app específico"""
    try:
        response = requests.get(
            f'{BASE_URL}/apps/details',
            params={'appId': app_id, 'store': store, 'country': country},
            headers={'Authorization': f'Bearer {API_KEY}'}
        )
        response.raise_for_status()
        
        app = response.json()['data']
        print(f'📋 Detalhes do App: {app["title"]}\n')
        print(f'   Developer: {app["developer"]}')
        print(f'   Score: ⭐ {app["score"]} ({app.get("ratings", "N/A")} avaliações)')
        print(f'   Installs: {app.get("installs", "N/A")}')
        print(f'   Version: {app.get("version", "N/A")}')
        print(f'   Size: {app.get("size", "N/A")}')
        print(f'   Updated: {app.get("updated", "N/A")}')
        print(f'   {app.get("summary", "")}')
        
        return app
    except requests.exceptions.RequestException as e:
        print(f'❌ Erro: {e}')

def get_similar_apps(app_id: str, store: str = 'playStore', num: int = 10):
    """Obter apps similares"""
    try:
        response = requests.get(
            f'{BASE_URL}/apps/similar',
            params={'appId': app_id, 'store': store, 'num': num},
            headers={'Authorization': f'Bearer {API_KEY}'}
        )
        response.raise_for_status()
        
        data = response.json()['data']
        print(f'🔗 Apps Similares a "{data["appId"]}":\n')
        
        for i, app in enumerate(data['results'], 1):
            print(f'{i}. {app["title"]}')
            print(f'   {app["developer"]} | ⭐ {app["score"]}')
            print('')
        
        return data
    except requests.exceptions.RequestException as e:
        print(f'❌ Erro: {e}')

if __name__ == '__main__':
    print('=== Apps Search API Examples ===\n')
    search_all_stores('whatsapp', num=3)

File: examples/nodejs/apps.js
const axios = require('axios');

const API_KEY = process.env.COGNIMA_API_KEY || 'ck_your_api_key';
const BASE_URL = 'https://cog.api.br/api/v1';

/**
 * Pesquisar apps em ambas as lojas (Google Play + App Store)
 */
async function searchAllStores() {
  try {
    const query = 'whatsapp';
    
    const response = await axios.get(
      `${BASE_URL}/apps/search`,
      {
        params: { 
          q: query,
          num: 5,
          country: 'br'
        },
        headers: {
          'Authorization': `Bearer ${API_KEY}`
        }
      }
    );
    
    const data = response.data.data;
    console.log(`📱 Pesquisa de Apps: "${data.query}"\n`);
    
    console.log('=== Google Play Store ===');
    data.playStore.forEach((app, i) => {
      console.log(`${i + 1}. ${app.title}`);
      console.log(`   Developer: ${app.developer}`);
      console.log(`   Score: ${app.score} | ${app.price}`);
      console.log(`   ID: ${app.appId}`);
      console.log('');
    });
    
    console.log('=== Apple App Store ===');
    data.appStore.forEach((app, i) => {
      console.log(`${i + 1}. ${app.title}`);
      console.log(`   Developer: ${app.developer}`);
      console.log(`   Score: ${app.score} | ${app.free ? 'Free' : app.price}`);
      console.log(`   ID: ${app.appId}`);
      console.log('');
    });
    
    return data;
  } catch (error) {
    console.error('❌ Erro:', error.response?.data || error.message);
  }
}

/**
 * Pesquisar apenas na Google Play Store
 */
async function searchPlayStore() {
  try {
    const query = 'games';
    
    const response = await axios.get(
      `${BASE_URL}/apps/playstore`,
      {
        params: { 
          q: query,
          num: 5,
          country: 'br',
          lang: 'pt'
        },
        headers: {
          'Authorization': `Bearer ${API_KEY}`
        }
      }
    );
    
    const data = response.data.data;
    console.log(`🤖 Google Play Store: "${data.query}"\n`);
    
    data.results.forEach((app, i) => {
      console.log(`${i + 1}. ${app.title}`);
      console.log(`   ${app.developer} | ⭐ ${app.score} | ${app.price}`);
      console.log(`   ${app.summary?.substring(0, 60)}...`);
      console.log('');
    });
    
    return data;
  } catch (error) {
    console.error('❌ Erro:', error.response?.data || error.message);
  }
}

/**
 * Pesquisar apenas na Apple App Store
 */
async function searchAppStore() {
  try {
    const query = 'music';
    
    const response = await axios.get(
      `${BASE_URL}/apps/appstore`,
      {
        params: { 
          q: query,
          num: 5,
          country: 'br'
        },
        headers: {
          'Authorization': `Bearer ${API_KEY}`
        }
      }
    );
    
    const data = response.data.data;
    console.log(`🍎 Apple App Store: "${data.query}"\n`);
    
    data.results.forEach((app, i) => {
      console.log(`${i + 1}. ${app.title}`);
      console.log(`   ${app.developer} | ⭐ ${app.score} | ${app.free ? 'Free' : app.price}`);
      console.log(`   Genres: ${app.genres?.join(', ')}`);
      console.log('');
    });
    
    return data;
  } catch (error) {
    console.error('❌ Erro:', error.response?.data || error.message);
  }
}

/**
 * Obter detalhes de um app
 */
async function getAppDetails() {
  try {
    const appId = 'com.whatsapp'; // WhatsApp no Play Store
    
    const response = await axios.get(
      `${BASE_URL}/apps/details`,
      {
        params: { 
          appId: appId,
          store: 'playStore',
          country: 'br'
        },
        headers: {
          'Authorization': `Bearer ${API_KEY}`
        }
      }
    );
    
    const app = response.data.data;
    console.log(`📋 Detalhes do App: ${app.title}\n`);
    console.log(`   Developer: ${app.developer}`);
    console.log(`   Score: ⭐ ${app.score} (${app.ratings} avaliações)`);
    console.log(`   Installs: ${app.installs}`);
    console.log(`   Version: ${app.version}`);
    console.log(`   Size: ${app.size}`);
    console.log(`   Android: ${app.androidVersionText}`);
    console.log(`   Updated: ${app.updated}`);
    console.log(`   ${app.summary}`);
    
    return app;
  } catch (error) {
    console.error('❌ Erro:', error.response?.data || error.message);
  }
}

/**
 * Obter apps similares
 */
async function getSimilarApps() {
  try {
    const appId = 'com.whatsapp';
    
    const response = await axios.get(
      `${BASE_URL}/apps/similar`,
      {
        params: { 
          appId: appId,
          store: 'playStore',
          num: 5
        },
        headers: {
          'Authorization': `Bearer ${API_KEY}`
        }
      }
    );
    
    const data = response.data.data;
    console.log(`🔗 Apps Similares a "${data.appId}":\n`);
    
    data.results.forEach((app, i) => {
      console.log(`${i + 1}. ${app.title}`);
      console.log(`   ${app.developer} | ⭐ ${app.score}`);
      console.log('');
    });
    
    return data;
  } catch (error) {
    console.error('❌ Erro:', error.response?.data || error.message);
  }
}

// Executar exemplos
console.log('=== Apps Search API Examples ===\n');
searchAllStores();

